### PR TITLE
Fix caching of message bubble sizes.

### PR DESCRIPTION
**Pull request by charlesmchen**
The cache for message bubble sizes has used hash values which were often zero, causing collisions.  I've switched to hashing the underlying interaction's uniqueId.

PTAL @michaelkirk

### CODE REVIEW

File: Signal/src/Models/OWSDisplayedMessage.h [hunk 1]
```diff
@@ -1,38 +0,0 @@
-//  Created by Dylan Bourgeois on 29/11/14.
-//  Copyright (c) 2014 Hexed Bits. All rights reserved.
-//  Portions Copyright (c) 2016 Open Whisper Systems. All rights reserved.
-
-#import "JSQMessageData.h"
```
Comment (charlesmchen) [inline Signal/src/Models/OWSDisplayedMessage.h]: This class wasn't used anywhere.

File: Signal/src/Models/TSMessageAdapaters/TSMessageAdapter.m [hunk 2]
```diff
@@ -77,9 +77,8 @@ - (instancetype)initWithInteraction:(TSInteraction *)interaction
 
     _interaction = interaction;
     _messageDate = interaction.date;
-    // TODO casting a string to an integer? At least need a comment here explaining why we are doing this.
```
Comment (michaelkirk) [inline Signal/src/Models/TSMessageAdapaters/TSMessageAdapter.m]: ah yes, I remember that TODO. =(

File: Signal/src/Models/TSMessageAdapaters/TSMessageAdapter.m [hunk 3]
```diff
@@ -354,11 +353,9 @@ - (NSString *)text {
 
 - (NSUInteger)messageHash
 {
-    if (self.isMediaMessage) {
-        return [self.mediaItem mediaHash];
```
Comment (michaelkirk) [inline Signal/src/Models/TSMessageAdapaters/TSMessageAdapter.m]: Is this going to properly invalidate the bubble size when a media message changes state?

e.g. when it goes from from "downloading" -> "downloaded"?

Comment (charlesmchen) [inline Signal/src/Models/TSMessageAdapaters/TSMessageAdapter.m]: Great point. Fixed.

**Pull request merged**